pkg/apis/core/validation/validation.go

```diff
@@ -1533,6 +1533,26 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 	return allErrs
 }
 
+// ValidatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
+// This also take a flag to validate the secret name against DNSSubDomainformat (<253char) and DNS1123Label (<63char) format
+// as its required by some PV sources today, ex: CSI. This flag can help on racheting validation of secretRef field.
+func ValidatePVSecretReference(secretRef *core.SecretReference, allowDNSSubDomainFormat bool, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if len(secretRef.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	} else {
+		if !allowDNSSubDomainFormat {
```

Comment:
I'd drop the flag. The flag name doesn't reflect what the code does. Keeping it may also trick people into thinking that the function validates DNSSubDomainFormat, which it doesn't.

Reply:
dropped for now,  will bring that while getting onto the next phase.. thanks.